core/math/random_pcg.h

```diff
@@ -126,10 +126,18 @@ class RandomPCG {
 	}
 
 	_FORCE_INLINE_ double randfn(double p_mean, double p_deviation) {
-		return p_mean + p_deviation * (cos(Math_TAU * randd()) * sqrt(-2.0 * log(randd()))); // Box-Muller transform
+		double temp = randd();
+		if (fabs(temp) < CMP_EPSILON) {
```

Comment:
If it's e.g. `- CMP_EPSILON/2`, adding `CMP_EPSILON` won't be enough to make it far enough from 0.

Reply:
Ah, I forgot the fact that CMP_EPSILON2 is precision used by double. I've changed it a bit - does this looks correct now?

Reply:
But what the point if `randd` generates the value in range of 0.0 - 1.0 and SIGN will always return positive sign?

Reply:
Ah, and SIGN suddenly receive 0.0 from `randd`, then the `temp` become 0, because 0 *  (temp_abs + CMP_EPSILON); will be 0.

Reply:
Well, why do you use `fabs()` if `randd()` is always positive? That's what caused my confusion.

Reply:
Hmm.. how true..